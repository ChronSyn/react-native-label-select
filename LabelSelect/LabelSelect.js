/**
 * Created by wangyi27 on 2017-01-03.
 */

import React, {Component, PropTypes} from 'react';
import {
  View,
  Text,
  Image,
  Modal,
  ScrollView,
  TouchableHighlight
} from 'react-native';
import Styles, {Color} from './LabelSelectStyle';

class LabelSelect extends Component {
  closeIcon = require('./img/close.png')
  constructor(props) {
    super(props);
    // 初始状态
    this.state = {
      isModalVisible: false,
      selectList: this.deepCopy(this.props.selectList)
    };
    this.toggleItem = this.toggleItem.bind(this);
    this.cancelSelect = this.cancelSelect.bind(this);
    this.confirmSelect = this.confirmSelect.bind(this);
    this.openModal = this.openModal.bind(this);
  }
  deepCopy(obj) {
    return JSON.parse(JSON.stringify(obj));
  }
  setModalVisible(isVisible) {
    this.setState({isModalVisible: isVisible});
  }
  cancelItem(item) {
    this.props.changeSelect(item, false);
    this.setState({selectList: this.deepCopy(this.props.selectList)});
  }
  cancelSelect() {
    this.setState({selectList: this.deepCopy(this.props.selectList)});
    this.setModalVisible(false);
  }
  confirmSelect() {
    this.props.confirmSelect(this.deepCopy(this.state.selectList));
    this.setModalVisible(false);
  }
  toggleItem(index) {
    const {selectAttr} = this.props;
    let list = this.state.selectList;
    list[index][selectAttr] = !list[index][selectAttr];
    this.setState({
      selectList: list
    });
  }
  openModal() {
    !this.props.disable && !this.props.readOnly && this.setModalVisible(true);
  }
  render() {
    const {selectAttr, selectList, disable, readOnly} = this.props;
    const selectedList = selectList.filter(item => item[selectAttr]);
    let selectedItems = selectedList.map((item, index) =>
      <View style={[Styles.selectedItem, disable ? Styles.disableColor : {}]}
        key={'select' + index}>
        <Text style={Styles.labelText}>{item.name}</Text>
        {!disable && !readOnly && <TouchableHighlight
          style={Styles.closeContainer}
          underlayColor="transparent"
          activeOpacity={0.5}
          onPress={() => {this.cancelItem(item);}}>
          <View>
            <Image
              style={Styles.closeIcon}
              source={this.closeIcon}
              resizeMode="cover"/>
          </View>
        </TouchableHighlight>}
      </View>
    );
    let modalItems = this.state.isModalVisible ? this.state.selectList.map((item, index) =>
      <TouchableHighlight
        key={'modalItem' + index}
        activeOpacity={0.5}
        underlayColor="transparent"
        onPress={() => {this.toggleItem(index);}}>
        <View style={[Styles.modalItem, index ? {} : {borderTopWidth: 0}]}>
          <Text>{item.name}</Text>
          <View style={[Styles.outerCircle, item[selectAttr] ? Styles.enableCircle : {}]}>
            {item[selectAttr] && <View style={Styles.innerCircle}></View>}
          </View>
        </View>
      </TouchableHighlight>
    ) : null;
    return (
      <TouchableHighlight
        style={[Styles.selectedContainer, disable ? Styles.disableColor : {}]}
        onPress={this.openModal}
        activeOpacity={disable || readOnly ? 1 : 0.8}
        underlayColor={disable ? Color.disableColor : 'transparent'}>
        <View style={Styles.selectedView}>
          {selectedItems}
          <Modal
            transparent={true}
            visible={this.state.isModalVisible}
            onRequestClose={this.cancelSelect}>
            <View style={{flex: 1}}>
              <TouchableHighlight
                style={Styles.modalMask}
                activeOpacity={1}
                underlayColor="transparent"
                onPress={this.cancelSelect}>
                <TouchableHighlight
                  activeOpacity={1}
                  underlayColor="transparent"
                  style={Styles.modalContainer}>
                  <View style={Styles.modal}>
                    <View style={Styles.title}><Text style={Styles.titleText}>标  签</Text></View>
                    <View style={Styles.scrollView}>
                      <ScrollView>
                        {modalItems}
                      </ScrollView>
                    </View>
                    <View style={Styles.buttonView}>
                      <TouchableHighlight
                        underlayColor="transparent"
                        style={Styles.modalButton}
                        onPress={this.cancelSelect}>
                        <Text style={Styles.buttonText}>取消</Text>
                      </TouchableHighlight>
                      <TouchableHighlight
                        underlayColor="transparent"
                        onPress={this.confirmSelect}
                        style={[Styles.modalButton, Styles.confirmButton]}>
                        <Text style={Styles.buttonText}>确定</Text>
                      </TouchableHighlight>
                    </View>
                  </View>
                </TouchableHighlight>
              </TouchableHighlight>
            </View>
          </Modal>
        </View>
      </TouchableHighlight>
    );
  }
}

LabelSelect.defaultProps = {
  selectList: [],
  selectAttr: 'isSelected',
  changeSelect: () => {},
  confirmSelect: () => {},
  disable: false,
  readOnly: false
};

LabelSelect.propTypes = {
  selectAttr: PropTypes.string,
  selectList: PropTypes.array,
  confirmSelect: PropTypes.func,
  changeSelect: PropTypes.func,
  readOnly: PropTypes.bool,
  enable: PropTypes.bool
};

export default LabelSelect;
