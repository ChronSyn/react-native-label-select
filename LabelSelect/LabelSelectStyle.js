/**
 * Created by wangyi27 on 2017-01-03.
 */

import {Dimensions, StyleSheet} from 'react-native';
const window = Dimensions.get('window');
const {width, height, scale} = window;

export const Color = {
  disableColor: '#eaeaea',
  main: '#40cca2'
};

export default StyleSheet.create({
  selectedItem: {
    margin: 4,
    borderWidth: 2 / scale,
    borderRadius: 4,
    borderColor: '#aaa',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fefefe'
  },
  selectedContainer: {
    padding: 4,
    borderWidth: 2 / scale,
    borderRadius: 4,
    borderColor: '#c8c8c8',
    backgroundColor: '#fcfcfc'
  },
  disableColor: {
    backgroundColor: Color.disableColor
  },
  selectedView: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    flexWrap: 'wrap'
  },
  labelText: {
    padding: 6,
    fontSize: 14,
    lineHeight: 14
  },
  closeContainer: {
    padding: 8,
    borderLeftWidth: 2 / scale,
    borderLeftColor: '#c8c8c8'
  },
  closeIcon: {
    width: 10,
    height: 10
  },
  modalMask: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#00000077'
  },
  modalContainer: {},
  modal: {
    height: height * 0.6,
    width: width * 0.6,
    overflow: 'hidden',
    borderRadius: 10,
    backgroundColor: '#fff'
  },
  title: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderBottomWidth: 2 / scale,
    borderBottomColor: '#bbb'
  },
  titleText: {
    fontSize: 18,
    lineHeight: 20
  },
  scrollView: {
    height: height * 0.6 - 80
  },
  buttonView: {
    height: 40,
    backgroundColor: Color.main,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  modalItem: {
    height: 50,
    paddingHorizontal: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderTopWidth: 0.5,
    borderTopColor: '#bbb'
  },
  buttonText: {
    color: '#fff',
    fontSize: 16
  },
  modalButton: {
    width: width * 0.3,
    paddingLeft: 20,
    paddingRight: 20,
    justifyContent: 'center',
    alignItems: 'center'
  },
  confirmButton: {
    borderLeftWidth: 0.5,
    borderLeftColor: '#fff'
  },
  outerCircle: {
    borderWidth: 2 / scale,
    borderColor: '#888',
    width: 20,
    height: 20,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center'
  },
  enableCircle: {
    borderColor: Color.main
  },
  innerCircle: {
    backgroundColor: Color.main,
    width: 16,
    height: 16,
    borderRadius: 8
  }
});
