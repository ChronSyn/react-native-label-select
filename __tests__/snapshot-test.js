import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import LabelSelect from '../LabelSelect';

import mock from './__mock__/mock';

const {Label, ModalItem} = LabelSelect;

let selectedItems = mock.selectedList.map((item, index) =>
  <Label
    key={'label-' + index}
    data={item}
    onCancel={() => {mock.selectedList.splice(index);}}>
    {item.text}
  </Label>
);

let otherItems = mock.list.map((item, index) =>
  <ModalItem
    key={'modal-item-' + index}
    data={item}>
    {item.text}
  </ModalItem>
);

// snapshot test

it('renders enabled LabelSelect', () => {
  let tree = renderer.create(
    <LabelSelect
      title="Test1"
      onConfirm={() => {}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders readOnly LabelSelect', () => {
  let tree = renderer.create(
    <LabelSelect
      readOnly={true}
      title="Test2"
      onConfirm={() => {}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders disabled LabelSelect', () => {
  let tree = renderer.create(
    <LabelSelect
      enable={false}
      title="Test3"
      onConfirm={() => {}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a modal item', () => {
  let item = mock.list[0];
  expect(renderer.create(
    <ModalItem data={item}>{item.text}</ModalItem>
  )).toMatchSnapshot();
});

// shallow test

it('interact with modal', () => {
  let arr = [];
  const item = mock.list[0];
  const tree = shallow(
    <LabelSelect
      title="Test1"
      onConfirm={(list) => {arr = list;}}
      >
      {selectedItems}
      {otherItems}
    </LabelSelect>
  );
  let select = tree.instance();
  select.openModal();
  expect(tree.state('isModalVisible')).toEqual(true);
  select.toggleSelect(item);
  select.confirmSelect();
  expect(arr[0]).toEqual(item);
  expect(tree.state('isModalVisible')).toEqual(false);
  select.openModal();
  select.cancelSelect();
  expect(tree.state('isModalVisible')).toEqual(false);
});

it('selecte a item', () => {
  let data = mock.list[0];
  let result;
  const tree = shallow(
    <ModalItem
      data={data}
      toggleSelect={(item) => {result = item;}}
      >
      {data.text}
    </ModalItem>
  );
  let instance = tree.instance();
  instance._toggleSelect();
  expect(result).toEqual(data);
});
