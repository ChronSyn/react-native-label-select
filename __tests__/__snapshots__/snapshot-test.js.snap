exports[`test renders a modal item 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
        }
      }>
      C
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#888",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 20,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 20,
          },
          Object {},
        ]
      } />
  </View>
</View>
`;

exports[`test renders disabled LabelSelect 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#eaeaea",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "padding": 6,
          },
          Object {
            "color": "#999",
          },
        ]
      }>
      A
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#eaeaea",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "padding": 6,
          },
          Object {
            "color": "#999",
          },
        ]
      }>
      B
    </Text>
  </View>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {},
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "height": 800.4,
                "overflow": "hidden",
                "width": 450,
              }
            }>
            <View
              style={
                Object {
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "lineHeight": 20,
                  }
                }>
                Test3
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 720.4,
                }
              }>
              <ScrollView />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#40cca2",
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "space-around",
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "width": 225,
                    },
                  ]
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }>
                  Cancel
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {
                        "borderLeftColor": "#fff",
                        "borderLeftWidth": 0.5,
                      },
                    ],
                  ]
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }>
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`test renders enabled LabelSelect 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "padding": 6,
          },
          false,
        ]
      }>
      A
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderLeftColor": "#c8c8c8",
            "borderLeftWidth": 1,
            "padding": 8,
          },
        ]
      }
      testID={undefined}>
      <View>
        <Image
          resizeMode="cover"
          source={1}
          style={
            Object {
              "height": 10,
              "width": 10,
            }
          } />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "padding": 6,
          },
          false,
        ]
      }>
      B
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderLeftColor": "#c8c8c8",
            "borderLeftWidth": 1,
            "padding": 8,
          },
        ]
      }
      testID={undefined}>
      <View>
        <Image
          resizeMode="cover"
          source={1}
          style={
            Object {
              "height": 10,
              "width": 10,
            }
          } />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#f6f6f6",
            "borderColor": "#aaa",
            "borderRadius": 6,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 4,
            "overflow": "hidden",
          },
          Object {
            "padding": 7,
          },
        ],
      ]
    }
    testID={undefined}>
    <Image
      resizeMode="cover"
      source={1}
      style={
        Object {
          "height": 12,
          "width": 12,
        }
      } />
  </View>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {},
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "height": 800.4,
                "overflow": "hidden",
                "width": 450,
              }
            }>
            <View
              style={
                Object {
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "lineHeight": 20,
                  }
                }>
                Test1
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 720.4,
                }
              }>
              <ScrollView />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#40cca2",
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "space-around",
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "width": 225,
                    },
                  ]
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }>
                  Cancel
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {
                        "borderLeftColor": "#fff",
                        "borderLeftWidth": 0.5,
                      },
                    ],
                  ]
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }>
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`test renders readOnly LabelSelect 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Object {},
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "padding": 6,
          },
          false,
        ]
      }>
      A
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f6f6f6",
          "borderColor": "#aaa",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 4,
          "overflow": "hidden",
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "lineHeight": 14,
            "padding": 6,
          },
          false,
        ]
      }>
      B
    </Text>
  </View>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {},
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 10,
                "height": 800.4,
                "overflow": "hidden",
                "width": 450,
              }
            }>
            <View
              style={
                Object {
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "lineHeight": 20,
                  }
                }>
                Test2
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 720.4,
                }
              }>
              <ScrollView />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#40cca2",
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "space-around",
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "width": 225,
                    },
                  ]
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }>
                  Cancel
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "width": 225,
                      },
                      Object {
                        "borderLeftColor": "#fff",
                        "borderLeftWidth": 0.5,
                      },
                    ],
                  ]
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }>
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
